pydatastructs/trees/m_ary_trees.py

```diff
@@ -0,0 +1,168 @@
+from pydatastructs.utils import M_AryTreeNode, BinomialTreeNode
+from pydatastructs.linear_data_structures.arrays import ArrayForTrees
+
+__all__ = [
+    'M_AryTree'
+]
+
+class M_AryTree(object):
```

Comment:
May be we can have an abstract `Tree` class from where all types of trees can inherit. Can you add such class in a new file `trees.tree.py` in a different PR and subclass in all kind of trees?

Reply:
Or may be adding such class to `misc_util.py` will also suffice. Adding new files for trivial purposes doesn't makes sense.

Reply:
Do you want me to add the Tree class in `utils/misc_util.py`?
That file has mostly just different kind of nodes.

Reply:
Yeah, that file contains utilities for other modules, it's fine add trivial abstract classes there which might be of use across various modules.

Reply:
Okay. I will work upon it in another PR.